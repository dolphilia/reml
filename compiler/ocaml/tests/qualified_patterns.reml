module test.qualified.patterns

let handle_option(value) =
  match value with
  | Option.Some(x) -> x
  | Option.None -> 0

let handle_dsl(node) =
  match node with
  | DSL.Node(tag) -> tag
  | DSL.Leaf -> 0

let inspect_record(record) =
  match record with
  | { status: Option.None, payload } -> payload
  | { status: Option.Some(value), .. } -> value
