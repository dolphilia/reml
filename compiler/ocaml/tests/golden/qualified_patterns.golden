module test.qualified.patterns
let handle_option(value) = match(var(value)) { Option.Some(x) => var(x) | Option.None => int(0:base10) }
let handle_dsl(node) = match(var(node)) { DSL.Node(tag) => var(tag) | DSL.Leaf => int(0:base10) }
let inspect_record(record) = match(var(record)) { { status: Option.None, payload } => var(payload) | { status: Option.Some(value); .. } => var(value) }
